src/execute/julia.ts

```diff
@@ -600,16 +600,21 @@ async function writeJuliaCommand(
   // but they could also end in a partial one.
   // so to read and process them all correctly, we read in a fixed number of bytes, if there's a newline, we process
   // the string up to that part and save the rest for the next round.
-  let restOfPreviousResponse = "";
+  let restOfPreviousResponse = new Uint8Array(512);
+  let restLength = 0; // number of valid bytes in restOfPreviousResponse
   while (true) {
-    let response = restOfPreviousResponse;
-    const newlineAt = response.indexOf("\n");
+    const respArray : Uint8Array[] = [];
+    let respLength = 0;
+    let response = "";
+    const newlineAt = restOfPreviousResponse.indexOf(10);
     // if we already have a newline, we don't need to read from conn
-    if (newlineAt !== -1) {
-      restOfPreviousResponse = response.substring(newlineAt + 1);
-      response = response.substring(0, newlineAt);
+    if (newlineAt !== -1 && newlineAt < restLength ) {
+      response = new TextDecoder().decode(restOfPreviousResponse.slice(0, newlineAt));
+      restOfPreviousResponse.set(restOfPreviousResponse.slice(newlineAt + 1, restLength));
+      restLength -= newlineAt+1;
```

Comment:
formatting issue here `newlineAt+1` -> `newlineAt + 1`.

We autoformat the code in VS code; could you run `deno fmt julia.ts` on your side? (if you don't have `deno` installed on your system, you can grab the one that Quarto installs at `package/dist/bin/tools/deno` in your quarto-cli repo)

Reply:
ok, I did run `deno fmt`